### ISSUE TITLE
How to deal with generated code via preBuildCommands?

### ISSUE THREAD

**Issue by AndrejMitrovic** [dlang/dub#2656]
I have what is essentially:

```sdl
preBuildCommands "./scripts/run-protobuf.sh"
sourcePaths "lib/generated"
```

Where the script runs the protobuf compiler and generates some D code within `lib/generated`.

But on the first invocation dub will emit a warning (plus a build error):

```
Warning Invalid source/import path: /c/dev/project/lib
```

Is there a better way to do this?

Because the current behavior is very unstable, essentially the build will fail the first time around if any of the project's code imports the generated code, but it will then succeed on the next dub rebuild as the `preBuildCommands` will have been executed by then and the import warnings will no longer exist.

Moved from https://github.com/vibe-d/vibe.d/issues/2733, filed there by accident.

**Comment by rikkimax** [comment]
That is odd, but try creating the lib directory manually and don't let the generator own it.

**Comment by rikkimax** [comment]
No, that's fine, I just think dub is doing some analysis a bit too soon and you're giving the generator the control to create a directory in which it's doing analysis on.

**Comment by AndrejMitrovic** [comment]
I see what you mean, make the dir and commit it empty first. Yeah, that can do the trick.
